components/ILIAS/Authentication/classes/Password/class.ilLocalPasswordSettingsGUI.php

```diff
@@ -0,0 +1,249 @@
+        $this->password_manager = ilLocalPasswordManager::getInstance();
+        $this->lng->loadLanguageModule('user');
+    }
+
+    public function executeCommand(): void
+    {
+        $cmd = $this->ctrl->getCmd();
+        switch ($cmd) {
+            default:
+                $this->$cmd();
+
+                break;
+        }
+    }
+
+    public function showPassword(
+        Form $form = null,
+        bool $hide_form = false
+    ): void {
+        // check whether password of user have to be changed
+        // due to first login or password of user is expired
+        if ($this->user->isPasswordChangeDemanded()) {
+            $this->tpl->setOnScreenMessage('info', $this->lng->txt('password_change_on_first_login_demand'));
+        } elseif ($this->user->isPasswordExpired()) {
+            $msg = $this->lng->txt('password_expired');
+            $password_age = $this->user->getPasswordAge();
+            $this->tpl->setOnScreenMessage('info', sprintf($msg, $password_age));
+        }
+
+        if (!$form && !$hide_form) {
+            $form = $this->getPasswordForm();
+        }
+        $this->tpl->setContent(!$hide_form ? $this->ui_renderer->render($form) : '');
+        $this->tpl->printToStdout();
+    }
+
+    public function getPasswordForm(
+        ServerRequestInterface $request = null,
+        array $errors = []
+    ): Form {
+        $items = [];
+        if ($this->password_manager->allowPasswordChange($this->user)) {
+            $pw_info_set = false;
+            if ((int) $this->user->getAuthMode(true) === ilAuthUtils::AUTH_LOCAL) {
+                $cpass = $this->ui_factory->input()->field()->password(
+                    $this->lng->txt('current_password'),
+                    ilSecuritySettingsChecker::getPasswordRequirementsInfo()
+                );
+
+                $pw_info_set = true;
+                if ($this->user->getPasswd()) {
+                    $cpass = $cpass->withRequired(true);
+                }
+                $cpass = $cpass->withRevelation(true);
+                $cpass_error = $errors['current_password'] ?? [];
+                if ($cpass_error !== []) {
+                    $cpass = $cpass->withError(implode('<br>', $cpass_error));
+                }
+
+                $items['current_password'] = $cpass;
+            }
+
+            // new password
+            $ipass = $this->ui_factory->input()->field()->password(
+                $this->lng->txt('desired_password'),
+            );
+            if ($pw_info_set === false) {
+                $ipass = $ipass->withByline(ilSecuritySettingsChecker::getPasswordRequirementsInfo());
+            }
+            $ipass = $ipass->withRequired(true);
+            $ipass = $ipass->withRevelation(true);
+            $ipass_error = $errors['new_password'] ?? [];
+            if ($ipass_error !== []) {
+                $ipass = $ipass->withError(implode('<br>', $ipass_error));
+            }
+            $items['new_password'] = $ipass;
```

Comment:
Could we add `withAdditionalTransformation` for the validations you currently do in `savePassword`?